pandas/core/generic.py

```diff
@@ -7660,6 +7618,231 @@ def nanptp(values, axis=0, skipna=True):
             ``numpy.ndarray`` method ``ptp``.""",
             nanptp)
 
+    _shared_docs['transform'] = ("""
+    Generate transformed same dimension Series or DataFrame.
+
+    .. versionadded:: 0.20.0
+
+    Parameters
+    ----------
+    func : Function, dictionary, or list of functions
+
+        Function: Function to apply to the DataFrame/Series.
+        ``args``, and ``kwargs`` are passed into ``func``.
+
+        Accepted Combinations are:
+
+        - function
+        - list of functions
+        - dict of column names -> functions (or list of functions).
+
+    args : Iterable, optional
+        Positional arguments passed into ``func``.
+
+        Example of accepted value:
+
+        - axis: 0 (default) to apply func to columns
+
+                 1 to apply func to rows.
+
+    kwargs : Mapping, optional
+        A dictionary of keyword arguments passed into ``func``.
+
+    Returns
+    -------
+    transformed : Series or DataFrame (same as input)
+
+    Notes
+    --------
+    The function performs the transformation.
+    It takes a Series as input and must return a same length Series.
+    Note that the function may modify the Series index.
+
+    If applied to a Series the function is applied once.
+    If applied to a DataFrame the function is applied on each row or column. 
+    Note that if the function modifies the index it must do so consistently
+    in the case of a DataFrame.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame(np.arange(30).reshape(10, 3),
+    ...                   columns=['A', 'B', 'C'],
+    ...                   index=pd.date_range('1/1/2000', periods=10))
+    >>> df.iloc[3:7] = np.nan
```

Comment:
Could you explain why you use missing values here? I'd prefer a shorter example (say 4-5 rows). The missing values don't seem to play a role, so I think they can be removed.

Reply:
The missing values were present in the [original example](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.transform.html). So I left them.  
They do not play a key role except maybe that nan do not crash transform.  
If you still want to remove them, I will.

Reply:
I think the expectation is that pandas methods gracefully handle NaN, so in this case removing them is good.

Reply:
**showing** that nan do not crash transform

Reply:
I removed NaN from the examples.